src/libraries/System.Runtime.Serialization.Schema/src/System/Runtime/Serialization/Schema/ContractCodeDomInfo.cs

```diff
@@ -0,0 +1,54 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.CodeDom;
+using System.Collections.Generic;
+using System.Xml;
+using System.Xml.Schema;
+
+using ExceptionUtil = System.Runtime.Serialization.DiagnosticUtility.ExceptionUtility;
+
+namespace System.Runtime.Serialization
+{
+    internal sealed class ContractCodeDomInfo
+    {
+        private string? _clrNamespace;
+        // NOTE TODO smolloy - This was a Dictionary<string,object?> previously, so adding a duplicate entry would throw an exception.
+        // HashSet does not allow duplicates either, but it just returns false instead of throwing. I think it's safe to not
```

Comment:
If you want to protect against this, you could add a helper method on this class to add a member name and have a Debug.Assert that _memberNames doesn't contain it. Then fix up the call points which call GetMemberNames().Add to use the helper method instead. But I agree with you, I think this will be find as you can't have 2 data members (property/fields) on a type with the same member name so it shouldn't be possible to get into that situation in the first place.